pkg/machines/components/create-vm-dialog/createVmDialog.jsx

```diff
@@ -508,60 +475,53 @@ const StorageRow = ({ connectionName, storageSize, storageSizeUnit, onValueChang
 
     return (
         <>
-            <label className="control-label" htmlFor="storage-pool-select">
-                {_("Storage")}
-            </label>
-            <CockpitSelect.Select id="storage-pool-select"
-                           initial={storagePoolName}
-                           onChange={e => onValueChanged('storagePool', e)}>
-                <CockpitSelect.SelectEntry data="NewVolume" key="NewVolume">{_("Create new volume")}</CockpitSelect.SelectEntry>
-                <CockpitSelect.SelectEntry data="NoStorage" key="NoStorage">{_("No storage")}</CockpitSelect.SelectEntry>
-                <CockpitSelect.SelectDivider />
-                <optgroup key="Storage pools" label="Storage pools">
-                    { storagePools.map(pool => {
-                        if (pool.volumes && pool.volumes.length)
-                            return <CockpitSelect.SelectEntry data={pool.name} key={pool.name}>{pool.name}</CockpitSelect.SelectEntry>;
-                    })}
-                </optgroup>
-            </CockpitSelect.Select>
+            <FormGroup label={_("Storage")} fieldId="storage-pool-select">
+                <CockpitSelect.Select id="storage-pool-select"
+                                      extraClass="pf-c-form-control"
+                                      initial={storagePoolName}
+                                      onChange={e => onValueChanged('storagePool', e)}>
+                    <CockpitSelect.SelectEntry data="NewVolume" key="NewVolume">
+                        {_("Create new volume")}
+                    </CockpitSelect.SelectEntry>
+                    <CockpitSelect.SelectEntry data="NoStorage" key="NoStorage">
+                        {_("No storage")}
+                    </CockpitSelect.SelectEntry>
+                    <CockpitSelect.SelectDivider />
+                    <optgroup key="Storage pools" label="Storage pools">
+                        { storagePools.map(pool => {
+                            if (pool.volumes && pool.volumes.length)
+                                return <CockpitSelect.SelectEntry data={pool.name} key={pool.name}>{pool.name}</CockpitSelect.SelectEntry>;
+                        })}
+                    </optgroup>
+                </CockpitSelect.Select>
+            </FormGroup>
 
             { storagePoolName !== "NewVolume" &&
             storagePoolName !== "NoStorage" &&
-            <>
-                <label className="control-label" htmlFor="storage-volume-select">
-                    {_("Volume")}
-                </label>
+            <FormGroup label={_("Volume")}
+                       fieldId="storage-volume-select"
+                       helperTextInvalid={_("This volume is already used by another VM.")}
+                       validated={isVolumeUsed[storageVolume] && isVolumeUsed[storageVolume].length > 0 ? "warning" : "default"}>
```

Comment:
Does this validation work? I tried to reproduce it locally and I cannot get this warning.

Reply:
Ah good catch - helperTextInvalid is only for error. For warning there are other properties.